### ISSUE TITLE
FireRainbow breaks Firebugs 'Add Watch'

### ISSUE THREAD

**Issue by MikeRatcliffe** [binaryage/firerainbow#6]
<pre>
Hi Antonin

I have had a problem in Firebug for a while now that if I highlight text<br>in the Script Panel, right-click and 'Add Watch' the wrong text is added<br>... I have just discovered that this issue is caused by<br>FireRainbow.

Here is a small test case so that you can see what is happening:
=================
&lt;html&gt;
&lt;head&gt;
&lt;script&gt;
    function doit() {
        var s = 'Just some dummy Text';
        debugger;
    }
&lt;/script&gt;
&lt;/head&gt;
&lt;body onload="doit();"&gt;
    1. Activate Firebug and refresh the page to trigger debugging
    2. When debugging starts, highlight the 's' in s += 'Just some dummy<br>Text';
    3. Right-click the highlighted 's' and choose 'Add Watch'
    4. If the correct text was added to the Watch Tab then right-click<br>the text and add watch again
    I just get a '&lt;' symbol added to the watch panel and a syntax<br>error ... strangely, it sometimes works but very rarely.
&lt;/body&gt;
&lt;/html&gt;
=================
What steps will reproduce the problem?
1. Install FireRainbow and restart Firefox
2. Load the test file
3. Activate Firebug and refresh the page to trigger debugging
4. When debugging starts, highlight the 's' in s += 'Just some dummy<br>Text';
5. Right-click the highlighted 's' and choose 'Add Watch'
6. If the correct text was added to the Watch Tab then right-click the<br>text and add watch again

What is the expected output?
I expect to see the highlighted text added to the Watch Panel

What do you see instead?
I just get "spa" added to the watch panel and syntax error.

Additional info:
This only happens when Firebug is installed
=================

Thanks for your hard work.

Mike Ratcliffe
</pre>

**Comment by darwin** [comment]
Yeah, firebug probably does not expect the content to be HTML, it expects it as plain text. I did on patch a year ago to make first firerainbow play nicely with FB, but maybe someone did other things and it broke.

I'll look at it some day. But it takes some time, because I've just fired my MBP.

**Comment by darwin** [comment]
please check this issue again with FireRainbow 1.2 (when released)
